main.cpp

```diff
@@ -1,38 +1,97 @@
 #include "main.h"
 #include <cmath>
 #include "rc.h"
+#include <commdlg.h> // za ChooseColor
 
 int number_dialog::idd() const {
 	return IDD_NUMBER;
 }
+
 bool number_dialog::on_init_dialog() {
+	set_int(IDC_EDIT1, number_of_circles);
 	return true;
 }
+
 bool number_dialog::on_ok() {
+	try {
+		number_of_circles = get_int(IDC_EDIT1);
+	}
+	catch (...) {
+		return false;
+	}
 	return true;
 }
 
 void main_window::on_paint(HDC hdc) {
+	RECT rc;
+	GetClientRect(*this, &rc);
+
+	int cx = (rc.left + rc.right) / 2;
+	int cy = (rc.top + rc.bottom) / 2;
+	int big_radius = min(rc.right - rc.left, rc.bottom - rc.top) / 3;

```

Comment:
Zašto `/ 3` da se veliki krug ne vidi cijeli? Ako stavite `/ 4` uvijek će biti vidljiv (i maksimalan).